src/dcs/inject.py

```diff
@@ -0,0 +1,160 @@
+from ghga_service_commons.auth.jwt_auth import JWTAuthContextProvider
+from hexkit.providers.akafka import KafkaEventPublisher, KafkaEventSubscriber
+from hexkit.providers.mongodb import MongoDbDaoFactory
+from typing_extensions import TypeAlias
+
+from dcs.adapters.inbound.event_sub import EventSubTranslator
+from dcs.adapters.inbound.fastapi_ import dummies
+from dcs.adapters.inbound.fastapi_.custom_openapi import get_openapi_schema
+from dcs.adapters.inbound.fastapi_.routes import router
+from dcs.adapters.outbound.dao import DrsObjectDaoConstructor
+from dcs.adapters.outbound.event_pub import EventPubTranslator
+from dcs.adapters.outbound.s3 import S3ObjectStorage
+from dcs.config import Config
+from dcs.core.auth_policies import WorkOrderContext
+from dcs.core.data_repository import DataRepository
+from dcs.ports.inbound.data_repository import DataRepositoryPort
+from dcs.utils import asyncnullcontext
+
+
+@asynccontextmanager
+async def prepare_core(*, config: Config) -> AsyncGenerator[DataRepositoryPort, None]:
+    """Constructs and initializes all core components and their outbound dependencies.
+    This is a context manager returns a container with all initialized resources upon
+    __enter__. The resources are teared down upon __exit__.
+
+    """
+    dao_factory = MongoDbDaoFactory(config=config)
+    drs_object_dao = await DrsObjectDaoConstructor.construct(dao_factory=dao_factory)
+    object_storage = S3ObjectStorage(config=config)
+
+    async with KafkaEventPublisher.construct(config=config) as event_pub_provider:
+        event_publisher = EventPubTranslator(config=config, provider=event_pub_provider)
+
+        yield DataRepository(
+            drs_object_dao=drs_object_dao,
+            object_storage=object_storage,
+            event_publisher=event_publisher,
+            config=config,
+        )
+
+
+OutboxCleaner: TypeAlias = Callable[[], Coroutine[None, None, None]]
+
+
+def get_configured_rest_app(*, config: Config) -> FastAPI:
+    """Create and configure a REST API application."""
+    app = FastAPI()
+    app.include_router(router)
+    configure_app(app, config=config)
+
+    def custom_openapi():
+        if app.openapi_schema:
+            return app.openapi_schema
+        openapi_schema = get_openapi_schema(app)
+        app.openapi_schema = openapi_schema
+        return app.openapi_schema
+
+    app.openapi = custom_openapi  # type: ignore [method-assign]
+
+    return app
+
+
+@asynccontextmanager
+async def prepare_rest_app(
+    *,
+    config: Config,
+    data_repo_override: Optional[DataRepositoryPort] = None,
+) -> AsyncGenerator[FastAPI, None]:
+    """Construct and initialize an REST API app along with all its dependencies.
+    By default, the core dependencies are automatically prepared but you can also
+    provide them using the data_repo_override parameter.
+    """
+    app = get_configured_rest_app(config=config)
+
+    data_repository_cm = (
```

Comment:
Maybe move this to its own function since it gets reused.
```suggestion
   data_repository_cm = resolve_context(config, data_repo_override)
```